### PR TITLE
[WFCORE-4442] Migrating UndertowHttpManagementService & HttpShutdownService to new MSC API

### PR DESCRIPTION
**Pull request by ropalka**
https://issues.jboss.org/browse/WFCORE-4442

### CODE REVIEW

File: server/src/main/java/org/jboss/as/server/mgmt/UndertowHttpManagementService.java [hunk 1]
```diff
@@ -237,17 +266,17 @@ public synchronized void start(StartContext context) throws StartException {
...
-        final NetworkInterfaceBinding interfaceBinding = interfaceBindingValue.getOptionalValue();
-        final NetworkInterfaceBinding secureInterfaceBinding = secureInterfaceBindingValue.getOptionalValue();
+        final SocketBinding basicBinding = socketBindingSupplier != null ? socketBindingSupplier.get() : null;
+        final SocketBinding secureBinding = secureSocketBindingSupplier != null ? secureSocketBindingSupplier.get() : null;
+        final NetworkInterfaceBinding interfaceBinding = interfaceBindingSupplier != null ? interfaceBindingSupplier.get() : null;
+        final NetworkInterfaceBinding secureInterfaceBinding = secureInterfaceBindingSupplier != null ? secureInterfaceBindingSupplier.get() : null;
         if (interfaceBinding != null) {
             useUnmanagedBindings = true;
-            final int port = portValue.getOptionalValue();
+            final int port = this.port != null ? this.port : -1;
```
Comment (bstansberry) [inline server/src/main/java/org/jboss/as/server/mgmt/UndertowHttpManagementService.java]: I think this null check can be changed to an assert this.port != null.  If an interfaceBinding is provided a port must be provided, and the intended user if it, org.jboss.as.host.controller.operations.HttpManagementAddHandler always provides a port.

File: server/src/main/java/org/jboss/as/server/mgmt/UndertowHttpManagementService.java [hunk 2]
```diff
@@ -237,17 +266,17 @@ public synchronized void start(StartContext context) throws StartException {
...
             if (port > 0) {
                 bindAddress = new InetSocketAddress(interfaceBinding.getAddress(), port);
             }
-            final int securePort = securePortValue.getOptionalValue();
+            final int securePort = this.securePort != null ? this.securePort : -1;
```
Comment (bstansberry) [inline server/src/main/java/org/jboss/as/server/mgmt/UndertowHttpManagementService.java]: Same comment as above re 'port'.

File: server/src/main/java/org/jboss/as/server/mgmt/UndertowHttpManagementService.java (hunk 1 above)
Comment (ropalka) [reply server/src/main/java/org/jboss/as/server/mgmt/UndertowHttpManagementService.java]: Updated accordingly @bstansberry

**Comment by bstansberry** [comment]
@jmesnil This LGTM.

**Pull request merged**